WebService/v1/Models/DeviceModelApiModel/DeviceModelSimulationScript.cs

```diff
@@ -26,9 +46,124 @@ public static DeviceModelSimulationScript FromServiceModel(Script value)
 
             return new DeviceModelSimulationScript
             {
-                Type = value.Type,
-                Path = value.Path
+                Type = string.IsNullOrEmpty(value.Type) ? null : value.Type,
+                Path = string.IsNullOrEmpty(value.Path) ? null : value.Path,
+                Params = value.Params
             };
         }
+
+        public bool IsEmpty()
+        {
+            return string.IsNullOrEmpty(this.Type)
+                   && string.IsNullOrEmpty(this.Path)
+                   && this.Params == null;
+        }
+
+        public void ValidateInputRequest(ILogger log)
+        {
+            const string INTERNAL = "internal";
+            const string NO_TYPE = "Simulation script type cannot be empty";
+            const string NO_PATH = "Simulation script path cannot be empty";
+
+            if (string.IsNullOrEmpty(this.Type))
+            {
+                log.Error(NO_TYPE, () => new { Script = this });
+                throw new BadRequestException(NO_TYPE);
+            }
+
+            if (string.IsNullOrEmpty(this.Path))
+            {
+                log.Error(NO_PATH, () => new { Script = this });
+                throw new BadRequestException(NO_PATH);
+            }
+
+            if (this.Type == INTERNAL)
+            {
+                this.ValidateParams(log);
+            }
+        }
+
+        /*
```

Comment:
Can this code be removed?

Reply:
Intend to give an example of json structure.